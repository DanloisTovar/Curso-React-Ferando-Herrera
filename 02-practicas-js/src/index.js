const saludar = function (nombre) {
    return `Hola ${nombre}`;
};

// console.log(saludar('Danlois'));

console.log(saludar);

const saludar2 = (nombre) => {
    return `Hola ${nombre}`;
};

console.log(saludar2);

const saludar3 = (nombre) => `Hola ${nombre}`;

console.log(saludar3);

const saludar4 = () => 'Hola Mundo!!!';

console.log(saludar4());

const getUser = () => ({
    uid: 'ABC123',
    username: 'El_Papi1502',
});

const user = getUser();

console.log(user);

// Tarea
// 1. Transformar a una función de Flecha
// 2. Tiene que retornar un objeto implícito
// 3. Pruebas

function getUsuarioActivo(nombre) {
    return {
        uid: 'ABC567',
        username: nombre,
    };
}

const usuarioActivo = getUsuarioActivo('Danlois');
console.log(usuarioActivo);

const getUsuarioActivo2 = (nombre) => ({
    uid: 'ABC567',
    username: nombre,
});
const usuarioActivo2 = getUsuarioActivo2('Danlois');
console.log(usuarioActivo2);
