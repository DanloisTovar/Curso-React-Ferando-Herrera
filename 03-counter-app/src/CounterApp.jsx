import { useState } from 'react';
import PropTypes from 'prop-types';

// component:
export const CounterApp = ({ value }) => {
    // hooks:
    const [counter, setCounter] = useState(value);

    // functions:
    const handleOnclickSum = () => {
        setCounter(counter + 1);
        console.log(counter);
    };

    const handleOnclickMinus = () => {
        setCounter(counter - 1);
        console.log(counter);
    };
    const handleOnclickReset = () => {
        setCounter(value);
        console.log(counter);
    };
    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button onClick={handleOnclickSum}>+1</button>
            <button onClick={handleOnclickMinus}>-1</button>
            <button onClick={handleOnclickReset}>Reset</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
    value: 0,
};
