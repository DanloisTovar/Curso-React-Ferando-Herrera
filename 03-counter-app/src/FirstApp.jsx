import PropTypes from 'prop-types';

export const FirstApp = ({ saludo, nombre, numero = 1 }) => {
    // console.log('saludo');

    return (
        <>
            <h1>{`${saludo} ${nombre}`}</h1>
            <p>Mi primera aplicaci√≥n</p>
            <p>suma: {numero + 1}</p>
        </>
    );
};

FirstApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    nombre: PropTypes.string.isRequired,
    numero: PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
    saludo: 'Hola NO soy',
    nombre: 'Goku',
    numero: 123,
};
