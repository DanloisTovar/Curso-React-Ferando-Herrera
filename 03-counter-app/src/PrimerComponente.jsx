export const PrimerComponente = () => {
    const imprimeNombre = (nombre) => {
        return `Hola soy ${nombre}`;
    };
    const nombreImpreso = imprimeNombre('Danlois');

    const arreglo = [1, 2, 3, 4, 5, 6, 7, 8, 9];

    const nuevoMensaje = {
        nombre: 'Danlois',
        edad: 29,
        ubicacion: 'Venezuela',
    };
    return (
        <>
            <h2>Primer Componente!!</h2>
            <p>Este es mi primer componente</p>
            <p>
                <strong>{nombreImpreso}</strong>
            </p>

            <p>lo que esta en el arreglo: {arreglo}</p>

            {arreglo
                ? arreglo.map((num) => {
                      return <p key={num}>{num}</p>;
                  })
                : null}

            <p>{nuevoMensaje.nombre}</p>
            <p>{nuevoMensaje.edad}</p>
            <p>{nuevoMensaje.ubicacion}</p>

            <code>{JSON.stringify(nuevoMensaje)}</code>
        </>
    );
};
