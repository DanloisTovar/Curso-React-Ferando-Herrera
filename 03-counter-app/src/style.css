html,
body {
    color: aqua;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

hr {
    border-color: aqua;
}

button {
    padding: 15px;
    font-size: 1.3rem;
}
