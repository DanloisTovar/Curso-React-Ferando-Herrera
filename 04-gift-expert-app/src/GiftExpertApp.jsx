

export const GiftExpertApp = () => {
    return (
        <>
            <h1>Aplicacion Gift</h1>
        </>
    );
};
