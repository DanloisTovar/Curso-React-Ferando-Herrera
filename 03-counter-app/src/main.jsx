import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { PrimerComponente } from './PrimerComponente';
import './style.css';
import { CounterApp } from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp nombre="Danlois" saludo="Hola Soy," numero={123} />

        <PrimerComponente />
        <br />
        <br />
        <hr />
        <CounterApp value={0} />
    </React.StrictMode>,
);
