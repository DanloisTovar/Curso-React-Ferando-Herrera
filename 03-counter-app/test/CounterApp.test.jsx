import { describe, expect, test } from '@jest/globals';
import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Pruebas en <CounterApp />', () => {
    test('Debe arrancar el test', () => {
        console.log('Arranque de test!');
    });
    test('debe mostrar <CounterApp /> correctamente con el valor por defecto', () => {
        const { getByText } = render(<CounterApp value={0} />);
        expect(getByText('0')).toBeTruthy();
    });

    test('Incrementar en "+1" con el boton click', () => {
        const { getByText } = render(<CounterApp />);
        fireEvent.click(getByText('+1'));
        expect(getByText('1')).toBeTruthy();
    });

    test('derecrementar en "-1" con el boton click', () => {
        const { getByText } = render(<CounterApp />);
        fireEvent.click(getByText('+1'));
        fireEvent.click(getByText('-1'));
        expect(getByText('0')).toBeTruthy();
    });

    test('resets counter to initial value on "Reset" button click', () => {
        const { getByText } = render(<CounterApp />);
        fireEvent.click(getByText('+1'));
        fireEvent.click(getByText('Reset'));
        expect(getByText('0')).toBeTruthy();
    });

    test('debería coincidir con la instantánea', () => {
        const { asFragment } = render(<CounterApp />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('debe mostrar <CounterApp /> correctamente con el valor por defecto', () => {
        render(<CounterApp value={0} />);
        expect(screen.getByText('0')).toBeTruthy();
    });

    test('Incrementar en "+1" con el boton click', () => {
        render(<CounterApp />);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('1')).toBeTruthy();
    });
});
