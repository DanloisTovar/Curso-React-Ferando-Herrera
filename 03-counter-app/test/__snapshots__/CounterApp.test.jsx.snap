// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debería coincidir con la instantánea 1`] = `
<DocumentFragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</DocumentFragment>
`;
